### PR TITLE
Add Test.Utils.Pretty and verifyMsg

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-970

### Overview

- Add a more descriptive `verifyMsg` function to the integration tests.
- Move the `Pretty` helper newtype into `Test.Utils.Pretty`.

### Comments

This code was rescued from #2708.

### CODE REVIEW

File: lib/core-integration/src/Test/Integration/Framework/DSL.hs [hunk 1]
```diff
@@ -2285,11 +2290,21 @@ wantedErrorButSuccess = liftIO
...
+-- If any of the assertions fail, 'a' is shown as the counter-example text.
+verify :: (Show a, MonadUnliftIO m) => a -> [a -> m ()] -> m ()
 verify a = counterexample msg . mapM_ (a &)
   where
-    msg = "While verifying " ++ show a
+    msg = "While verifying value:\n"+|indentF 2 (pShowBuilder a)|+""
+
+-- | Applies the value 'a' to all assertions in the given sequence.
+--
+-- Like 'verify', but the counterexample shows a description of what conditions were being checked.
```
Comment (jonathanknowles) [inline lib/core-integration/src/Test/Integration/Framework/DSL.hs]:
```suggestion
-- Like 'verify', but the counterexample shows a description of what conditions 
-- were being checked.
```

**Review by jonathanknowles — APPROVED** [review]
This looks good to me.

Just a minor suggestion.

File: lib/core-integration/src/Test/Integration/Framework/DSL.hs (hunk 1 above)
Comment (rvl) [reply lib/core-integration/src/Test/Integration/Framework/DSL.hs]: Oops, fixed.

File: lib/core-integration/src/Test/Integration/Scenario/CLI/Shelley/HWWallets.hs [hunk 2]
```diff
@@ -125,10 +127,10 @@ spec = describe "SHELLEY_CLI_HW_WALLETS" $ do
         eventually "Wallet balance is as expected" $ do
             Stdout og <- getWalletViaCLI ctx $ T.unpack (wDest ^. walletId)
             jg <- expectValidJSON (Proxy @ApiWallet) og
-            expectCliField (#balance . #available)
-                (`shouldBe` Quantity amount) jg
-            expectCliField (#balance . #total)
-                (`shouldBe` Quantity amount) jg
+            verify jq
```
Comment (Anviking) [inline lib/core-integration/src/Test/Integration/Scenario/CLI/Shelley/HWWallets.hs]:
```suggestion
            verify jg
```

**Pull request merged**